.github/workflows/cicd-main.yml

```diff
@@ -28,6 +36,60 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  pre-flight:
+    runs-on: ubuntu-latest
+    outputs:
+      test_to_run: ${{ steps.test_to_run.outputs.main }}
+      all: ${{ steps.all.outputs.main }}
+    steps:
+      - name: Parse test_to_run
+        id: test_to_run
+        run: |
+          parsed_string=$(echo ${{ inputs.test_to_run || 'all' }} | jq -c --raw-input 'split(",")')
+          echo "main=${parsed_string}" | tee -a "$GITHUB_OUTPUT"
+      - name: Parse all
+        id: all
+        run: |
+          echo "main=${{ contains(fromJSON(steps.test_to_run.outputs.main), 'all') }}" | tee -a "$GITHUB_OUTPUT"
+  
   build-container:
     if: ${{ github.event.label.name == 'Run CICD' || github.ref == 'refs/heads/main' }}
     uses: ./.github/workflows/_build_container.yml
+  
+  Unit_Tests:
+    needs: [build-container, pre-flight]
+    uses: ./.github/workflows/_run_test.yml
+    if: contains(fromJSON(needs.pre-flight.outputs.test_to_run), 'unit') || needs.pre-flight.outputs.all == 'true'
+    with:
+      IS_OPTIONAL: true
+      RUNNER: self-hosted-azure
+      TIMEOUT: 20
+      SCRIPT: |
+        nvidia-smi
+        cd ${ALIGNER_REPO_DIR}
+        torchrun --nproc_per_node 2 -m pytest . -rA -s -x
+
+  Functional_Tests:
+    name: ${{ matrix.TEST_NAME }}
+    needs: [build-container, pre-flight]
+    uses: ./.github/workflows/_run_test.yml
+    if: contains(fromJSON(needs.pre-flight.outputs.test_to_run), 'functional') || needs.pre-flight.outputs.all == 'true'
+    strategy:
+     matrix:
+       include:
+        # ALIGNER_CI_DIR/ALIGNER_REPO_DIR is available in the shell env
+        - TEST_NAME: ppo
+          SCRIPT: |
+            RM_NEMO_FILE=${ALIGNER_CI_DIR}/checkpoints/llama3--nlayers4-hidden64-ffn224-dummy_rm-megatron_gpt.nemo \
+            ACTOR_NEMO_FILE=${ALIGNER_CI_DIR}/checkpoints/tiny-llama3-results-nlayers2-hidden128-ffn448-nhead4-qgroup2-megatron_gpt.nemo \
+            bash ${ALIGNER_REPO_DIR}/tests/functional/ppo.sh
```

Comment:
WDYT about storing this shell script separately in the repo? That would allow running/testing it easily on other infra too

Reply:
addressed in https://github.com/NVIDIA/NeMo-Aligner/pull/356